znai-utils/src/main/java/com/twosigma/znai/utils/FilePathUtils.java

```diff
@@ -0,0 +1,18 @@
+package com.twosigma.znai.utils;
+
+import java.nio.file.Path;
+
+public class FilePathUtils {
+    private FilePathUtils() {
+    }
+
+    public static String fileNameWithoutExtension(Path path) {
+        String fileName = path.getFileName().toString();
+        int lastDotIdx = fileName.lastIndexOf('.');
```

Comment:
FilenameUtils.removeExtension() can do the same thing.
If you use a filename as String, you can remove the util class, too.